internal/services/keyvault/key_vault_secret_data_source.go

```diff
@@ -35,6 +35,12 @@ func dataSourceKeyVaultSecret() *pluginsdk.Resource {
 				ValidateFunc: keyVaultValidate.VaultID,
 			},
 
+			"key_vault_uri": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ValidateFunc: keyVaultValidate.VaultURI,
```

Comment:
since only 1 of these can be specified we should use the "AtLeastOneOf" validation to ensure this

Reply:
@katbyte thanks for the review! At first, I thought of going the way you're describing as, indeed, when you have key vault URI at hand, that's enough to pull a secret. Though, the resource exports many fields that depend on `key_vault_id`, e.g.:

```go
d.Set("resource_id", parse.NewSecretID(keyVaultId.SubscriptionId, keyVaultId.ResourceGroup, keyVaultId.Name, respID.Name, respID.Version).ID())
d.Set("resource_versionless_id", parse.NewSecretVersionlessID(keyVaultId.SubscriptionId, keyVaultId.ResourceGroup, keyVaultId.Name, respID.Name).ID())
```

So, I thought that even though `key_vault_id` seems redundant, it's a toll to pay for outputs consistency. So, a user is expected to pass either `key_vault_id` or both `key_vault_id` and `key_vault_uri`, but not `key_vault_uri` alone.
I should have added such comment to the PR to share more details on the implementation. :)